src/VisualStudio/CSharp/Impl/LanguageService/CSharpHelpContextService.cs

```diff
@@ -352,6 +352,16 @@ private static bool TryGetTextForKeyword(SyntaxToken token, ISyntaxFactsService
                 }
             }
 
+            if (token.IsKind(SyntaxKind.ClassKeyword)) {
+                if (token.Parent is ClassOrStructConstraintSyntax) {
+                    text = Keyword("classconstraint");
+                    return true;
+                } else {
+                    text = Keyword("class");
+                    return true;
+                }
+            }
```

Comment:
can we do the same for struct/new as those are also constraints.

Reply:
@TheSench feel free to include them in this PR, or not. They are called out in https://github.com/dotnet/docs/issues/20799 so we're not going to forget.

They would go to the same file as your current docs PR is changing, so there might be some sense in at least doing that side of things.

Reply:
Seeing as the build failed due to what appears to be a transient issue, I'll add this and kick off another one.  Is there a benefit to creating a new f1_keyword versus just using the existing `whereconstraint_CSharpKeyword` one for the constraint form of both of these?

Reply:
I'm wondering if we want to introduce a switch on `token.RawKind` here now that the number of special cases has grown a bunch.  I'll probably be tackling more of these cases in the next week, so that would be something I consider in the next one, but I'd be interested in direction before I get going on it, if possible.

Reply:
this code is C# specific.  So you don't need RawKind (you can use Kind()).  I wouldn't really do a switch here.  I think the code is clear when it handles each case specifically.  There's no need for brevity/perf as this will execute once only on hte rare case of someone hitting f1.

Reply:
Thanks for the extra context.